### PR TITLE
support vscode.dev link generation in notebook editor

### LINKED ISSUE #153944: Support notebooks in share vscode.dev link command
Re #153319

The notebook editors support deep links so if we can enable **Share vscode.dev link** command in notebooks, we can generate links for our endgame and my work notebooks and share with the team. The links to vscode.dev would be more approachable than finding notebooks in vscode repo on desktop or going through github.com links.

### PR DESCRIPTION
**Pull request by rebornix**
Fix #153944

Generate notebook deep links when the active editor is a notebook editor. The command is available in both cell text editor (monaco)'s context menu, cell title toolbar and cell context menu (right click on whitespace on a cell).

### CODE REVIEW

File: src/vs/workbench/contrib/notebook/browser/controller/coreActions.ts [hunk 1]
```diff
@@ -427,3 +428,9 @@ MenuRegistry.appendMenuItem(MenuId.EditorContext, {
 	group: CellOverflowToolbarGroups.Insert,
 	when: NOTEBOOK_EDITOR_FOCUSED
 });
+
+MenuRegistry.appendMenuItem(MenuId.NotebookCellTitle, {
+	title: localize('miShare', "Share"),
+	submenu: MenuId.MenubarShare,
```
Comment (alexr00) [inline src/vs/workbench/contrib/notebook/browser/controller/coreActions.ts:434]: A notebook cell is in the editor. Does the `EditorContextShare` menu id make more sense here?

Comment (rebornix) [reply src/vs/workbench/contrib/notebook/browser/controller/coreActions.ts:434]: Yep, `EditorContextShare` is better here.

**Comment by rebornix** [comment]
@alexr00 thanks for the review, code is now updated per comment. `EditorContextShare` works great.

image

**Review by alexr00 — APPROVED** [review]

**Comment by alexr00** [comment]
Thank you for the PR!

**Pull request merged**